shell/src/main/java/alluxio/cli/fs/command/AbstractDistributedJobCommand.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.cli.fs.command;
+
+import alluxio.ClientContext;
+import alluxio.cli.fs.command.job.JobAttempt;
+import alluxio.client.file.FileSystemContext;
+import alluxio.client.job.JobMasterClient;
+import alluxio.job.wire.Status;
+import alluxio.worker.job.JobMasterClientContext;
+
+import com.google.common.collect.Lists;
+
+import java.util.List;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.stream.Collectors;
+
+/**
+ * The base class for all the distributed job based {@link alluxio.cli.Command} classes.
+ * It provides handling for submitting multiple jobs and handling retries of them.
+ */
+public abstract class AbstractDistributedJobCommand extends AbstractFileSystemCommand {
+  private static final int DEFAULT_ACTIVE_JOBS = 1000;
+
+  protected List<JobAttempt> mSubmittedJobAttempts;
+  protected int mActiveJobs;
+  protected final JobMasterClient mClient;
+
+  protected AbstractDistributedJobCommand(FileSystemContext fsContext) {
+    super(fsContext);
+    mSubmittedJobAttempts = Lists.newArrayList();
+    final ClientContext clientContext = mFsContext.getClientContext();
+    mClient = JobMasterClient.Factory.create(
+        JobMasterClientContext.newBuilder(clientContext).build());
+    mActiveJobs = DEFAULT_ACTIVE_JOBS;
+  }
+
+  protected void drain() {
+    while (!mSubmittedJobAttempts.isEmpty()) {
+      waitJob();
+    }
+  }
+
+  /**
+   * Waits for at least one job to complete.
+   */
+  protected void waitJob() {
+    AtomicBoolean removed = new AtomicBoolean(false);
+    while (true) {
```

Comment:
Is it intentional that this while loop is very tight? Can we add a sleep in here to avoid too much unnecessary cpu?